RELEASING.md

```diff
@@ -0,0 +1,241 @@
+# Versioning and release processes
+
+## Versioning Process
+
+Ledger team has fully adopted the [PVP versioning scheme](https://pvp.haskell.org/) for
+all packages in this repo.
+
+Updating the version for every package in this repository is the responsinility of the
+developer that is introducing a change to any package that is subject to release
+process. Version must be updated in both the `CHANGELOG.md` and in the `.cabal` file for
+the package in the same PR, as well as every other package that is affected by the change
+in the repository. Every affected package that is subject to release process must receive
+a lower and, potentially, an upper bound update on the dependency that has experienced the
+change.
+
+For example if I add a function to `cardano-ledger-core` and use it in
+`cardano-ledger-conway`, then I must add a minor version bump to `cardano-ledger-core` and
+I also must add a lower version restriction in `cardano-ledger-conway` for
+`cardano-ledger-core`, as per PVP.
+
+In other words, when it comes to versioning, the fact that all of the packages in this
+repository are build together at the same time should be ignored. It should be assumed
+that every package can be released to CHaPs by itself at any point in time. This is
+necessary to guarantee that if such a package is compatible with other packages from this
+repository, which have been previously released, the version bounds should reflect that.
+
+### Upper bounds
+
+Normally, [strict upper bounds are required by
+PVP](https://pvp.haskell.org/#dependencies-in-cabal) for all dependencies of a
+package. This is the only exception we make to the PVP, and we do not require strict upper
+bounds for dependenicies that come from Hackage. For exmaple there is no need to add an
+upper bound on `containers` package, because that's just dumb. On the other hand all of
+the dependeices that come from CHaPs should follow strict bounds, because those packages
+are a lot more volatile and creating revisions for packages on CHaPs is not as smooth as
+it is on Hackage.
+
+### `CHANGELOG.md`
+
+Every package, that [is subject to the release
+process](packages-excluded-from-release-process), will always have the top most entry in
+the `CHANGELOG.md` set to an unreleased version. That top most section will have either
+the minor or the major versions bumped, but not both, when compared to the latest released
+version. Which one it will be depends on the recent changes that were added after the
+latest release was made. More on why this should always be true is in the [Release
+Process](#release-process) section. For example, if the latest version of
+`cardano-ledger-core` that was released to CHaPs is `1.20.1.0`, then there will be a
+section like this:
+
+```
+# Version history for `cardano-ledger-core`
+
+## 1.20.2.0
+
+* Add ...
+...
+```
+
+It is quite common to experience conflicts in the changelog, since that will be the most
```

Comment:
You might enjoy using `scriv`, we've been using it fairly happily in `plutus` to solve this.

Reply:
Nah, `scriv` is really a hammer that we don't need. It would be very useful if we were dealing with bundles as Plutus or Consensus does, but we don't have this problem. Moreover, we as developers, deal with git conflicts on daily basis, so couple more conflicts in the changelog is not a big deal.